prow/cmd/webhook-server/clients.go

```diff
@@ -0,0 +1,184 @@
+
+type GCPClient struct{
+	client *secretmanager.Client
+}
+
+func newGCPClient(client *secretmanager.Client) *GCPClient {
+	return &GCPClient{
+		client: client,
+	}
+}
+
+func (g *GCPClient) CreateSecret(ctx context.Context, secretID string) error {
+	_, err := g.client.CreateSecret(ctx, secretID)
+	if err != nil {
+		return fmt.Errorf("could not create secret %v", err)
+	}
+	return nil
+}
+
+func (g *GCPClient) AddSecretVersion(ctx context.Context, secretName string, payload []byte) error {
+	if err := g.client.AddSecretVersion(ctx, secretName, payload); err != nil {
+		return fmt.Errorf("could not add secret data %v", err)
+	}
+	return nil
+}
+
+func (g *GCPClient) GetSecretValue(ctx context.Context, secretName string, versionName string) ([]byte, error) {
+	exist, err := g.CheckSecret(ctx, secretName)
+    if err != nil {
+        return nil, err
+    }
+    if !exist {
+        return nil, errors.New("secret not exist")
+    }
+	return g.client.GetSecretValue(ctx, secretName, versionName)
+}
+
+func (g *GCPClient) CheckSecret(ctx context.Context, secretName string) (bool, error) {
+	var found bool
+	res, err := g.client.ListSecrets(ctx)
+	if err != nil {
+		return false, fmt.Errorf("could not make call to list secrets successfully %v", err)
+	}
+	for _, secret := range(res) {
+		if strings.Contains(secret.Name, secretID) {
+			found = true
+			break
+		}
+	}
+	return found, nil 
+}
+
+
+//for integration testing purposes. Not to be used in prod
+type localFSClient struct {
+	path   string
+	expiry int
+	dns    []string
+}
+func NewLocalFSClient(path string, expiry int, dns []string) *localFSClient {
+	return &localFSClient{
+		path: path,
+		expiry: expiry,
+		dns: dns,
+	}
+}
+
+func (l *localFSClient) CreateSecret(ctx context.Context, secretID string) (error) {
+	if _, err := os.Stat(l.path); errors.Is(err, os.ErrNotExist) {
+		err := os.Mkdir(l.path, 0755)
+		if err != nil {
+			return fmt.Errorf("unable to create secret dir %v", err)
+		}
+	}
+	return nil
+}
+
+func (l *localFSClient) AddSecretVersion(ctx context.Context, secretName string, payload []byte) error {
+	certFile := filepath.Join(l.path, certFile)
+	privKeyFile := filepath.Join(l.path, privKeyFile)
+	caBundleFile := filepath.Join(l.path, caBundleFile)
+
+	serverCertPerm, serverPrivKey, caPem, _, err := genSecretData(l.expiry, l.dns)
+	if err != nil {
+		return err
+	}
+	if err := ioutil.WriteFile(certFile, []byte(serverCertPerm), 0666); err != nil {
+		return fmt.Errorf("could not write contents of cert file")
+	}
+	if err := ioutil.WriteFile(privKeyFile, []byte(serverPrivKey), 0666); err != nil {
+		return fmt.Errorf("could not write contents of privkey file")
+	}
+	if err := ioutil.WriteFile(caBundleFile, []byte(caPem), 0666); err != nil {
+		return fmt.Errorf("could not write contents of caBundle file")
+	}
+	return nil
+}
+
+func (l *localFSClient) GetSecretValue(ctx context.Context, secretName, versionName string) ([]byte, error) {
+	var fileNames []string
```

Comment:
I am not sure we need another list for fileNames and have another for loop to get the fileNames. Can't we check if len(files)<2 and then do 
```
for _, f := range files {
  fileName := f.Name()
  ...
  }